### ISSUE TITLE
React will fail to compile if you install date-fns prior to installing material-table

### ISSUE THREAD

**Issue by afenrir** [mbrn/material-table#220]
I am using `date-fns` in my project independently of `material-table`. I had installed `date-fns` before installing `material-table`. The current version of `date-fns` is 1.30.1. When I later installed `material-table`, importing it into my project caused a long list of import errors like this:

```
Module not found: Can't resolve 'date-fns/addDays' in '/{project-root}/node_modules/material-ui-pickers/utils/date-fns-utils'
```

On further investigation, it appears that your dependency, `material-ui-pickers`, requires `date-fns@2.0.0-alpha.16`, which is the next version of `date-fns`. The next version of `date-fns` uses different paths to the current version, so when you install the current version first, the compiler will try to use that instead of the next version. 

I'm not entirely sure what you can do about this, as it is really a problem with one of your dependencies and how npm resolves conflicting dependencies. However, the problem can be fixed if the user installs `date-fns@next` instead. Note there are other breaking changes to that library in the next version; I had to change my code. 

May I suggest that you add a note to the npm output on installation? It would have saved me about 20 minutes of investigation, and may prevent others giving up altogether. Perhaps the following note could include the version, because when I initially saw this my response was "yes":

```
Have you installed one of peer libraries?
 > date-fns 
 > luxon 
 > moment
```

**To Reproduce**
Simply install `date-fns` as a project dependency prior to installing `material-table`.

**Comment by jaimealv994** [comment]
the same happened to me. @afenrir  Have you managed to solve this problem?

**Comment by mbrn** [comment]
I upgraded all date-fns and date picker dependencies. This should be resolved with v1.23.2
